### PR TITLE
[Mellanox] Update Mellanox-SN5600-C256S1 SKU name and update buffers files

### PR DESCRIPTION
**Pull request by noaOrMlnx**
#### Why I did it
Update C256S1 SKU name and buffers calculations.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256S1/buffers_defaults_t0.j2 [hunk 1]
```diff
@@ -15,10 +16,10 @@
...
 {% set default_cable = '40m' %}
-{% set ingress_lossless_pool_size =  '118249472' %}
-{% set ingress_lossless_pool_xoff =  '15728640' %}
+{% set ingress_lossless_pool_size =  '115245056' %}
+{% set ingress_lossless_pool_xoff =  '14826496' %}
 {% set egress_lossless_pool_size =  '158229504' %}
-{% set egress_lossy_pool_size =  '118249472' %}
+{% set egress_lossy_pool_size =  '115245056' %}
 
 {% import 'buffers_defaults_objects.j2' as defs with context %}
```
Comment (r12f) [inline device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256S1/buffers_defaults_t0.j2:24]: will this config create a single lossy buffer pool?

Comment (noaOrMlnx) [reply device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256S1/buffers_defaults_t0.j2:24]: yes, it will create a single egress & ingress lossy pool

**Review by r12f — APPROVED** [review]
basic cable length and buffer pool setting both looks ok. approved.

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256S1/buffers_defaults_objects.j2
Comment (r12f) [inline device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256S1/buffers_defaults_objects.j2:1]: do we need to update the default object file to get the right buffer pool generated?

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256S1/buffers_defaults_t0.j2 (hunk 1 above)
Comment (r12f) [reply device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-C256S1/buffers_defaults_t0.j2:24]: hi Noa, just to double confirm - will the lossless pool be removed?

**Review by liat-grozovik — APPROVED** [review]

**Comment by noaOrMlnx** [comment]
@bingwang-ms can you please help to cherry-pick to 202405?

**Pull request merged**